src/app

```diff
@@ -218,6 +203,20 @@ case "$STATE" in
         ;;
 
     ArtifactInstall)
+        # Ensure PERSISTENT_STORE exists
+        mkdir -p "$PERSISTENT_STORE"
+        set +e
+        if touch "$PERSISTENT_STORE"/.rw_test; then
+            rm -f "$PERSISTENT_STORE"/.rw_test
+            rc=$?
+            if test $rc -ne 0; then
+                exit $rc
+            fi
+            set -e
+        else
+            echo "ERROR: cant write to persistent_store in $PERSISTENT_STORE" 1>&2
+            exit 1
+        fi
```

Comment:
okey, I will swallow trap removal, under the following conditions:
* put those lies into `assert_access` function (naming up to you)
* echo with ERROR if we failed to remove the file